composer/trainer/trainer.py

```diff
@@ -1046,6 +1046,12 @@ def __init__(
         self._ddp_sync_strategy = _get_ddp_sync_strategy(ddp_sync_strategy, self._find_unused_parameters)
         # Configure Deepspeed
         if self.state.deepspeed_config is not None:
+            for callback in self.state.callbacks:
+                if isinstance(callback, GradMonitor):
+                    raise TypeError('GradMonitor is not supported with DeepSpeed because DeepSpeed clears '
+                                    'the gradients before in the last call to .backward see: '
+                                    'https://github.com/microsoft/DeepSpeed/issues/2329 for more details.')
+
```

Comment:
```suggestion
            for callback in self.state.callbacks:
                if isinstance(callback, GradMonitor):
                    raise ValueError('GradMonitor is not supported with DeepSpeed because DeepSpeed clears '
                                    'the gradients before in the last call to .backward see: '
                                    'https://github.com/microsoft/DeepSpeed/issues/2329 for more details.')

```
For incompatible trainer args, we generally raise ValueErrors (for example, see auto grad accum + profiler)